### ISSUE TITLE
Offer validation

### ISSUE THREAD

**Issue by ZajdaBear** [allegro/allegro-api#8115]
Hi, I was hoping you could help me with a few questions. I am looking to improve the listing process, and it is crucial for me to know the exact state of an offer in the Allegro system.

1) Is there a way to trigger offer validation through the API?
The GET request https://api.allegro.pl/sale/product-offers/14082271300 shows no validation errors. However, when I try to PATCH the offer, I encounter a "MissingRequiredParameters" error. I am aware of the 'unfilled-parameters' resource, but I need to be able to work with DRAFT offers, which, as I was informed, are not included in this resource.

2) While using the mentioned 'unfilled-parameters' resource, does it also show invalid offer setups such as missing/invalid shipping rates, return policies, etc.? I assume not :)

To summarize, I need a way to perform a full validation of an offer, regardless of whether it is INACTIVE, ACTIVE, or ENDED. I want to achieve this in the same manner as you do it on the edit offer page ([https://allegro.pl.allegrosandbox.pl/offer/{offerId}](https://allegro.pl.allegrosandbox.pl/offer/%7BofferId%7D)) in your GUI. On this page, I trigger validation and can see missing parameters, disregarding the offer status, as well as any other issues with the offer.

I hope there is a solution :)

Thanks!

**Comment by AureliuszBrussy** [comment]
1. We won't trigger the offer validation other than when you try to edit it, so you must send a PATCH request with some changes. 

2. We will only return unfilled required parameters.

**Comment by ZajdaBear** [comment]
Hi, thanks for your reply!

I was trying to push a validation on offer, but no luck. I hope you can help me further!

Details:

GET https://api.allegro.pl/sale/product-offers/13591833929
trace-id: 94539581690cf7df
response: 
[response1.json](https://github.com/allegro/allegro-api/files/13645406/response1.json)
 - publication.status: ACTIVE
 - validation.errors [empty]
 - validatedAt: 2023-08-28T13:22:14.263Z

I know, the offer has missing required parameters, which I have checked using
GET https://api.allegro.pl/sale/offers/unfilled-parameters?parameterType=REQUIRED
trace_id: c3f0377f77d78af0

I try to edit an offer, to trigger validations, as adviced using PATCH https://api.allegro.pl/sale/product-offers/13591833929

patching product qty
trace-id: dd2f66ad011fd7e9
```{
    "stock": {
        "available": 13,
        "unit": "UNIT"
    }
}
```

response 
```{
    "errors": [
        {
            "code": "ConstraintViolationException.MissingRequiredParameters",
            "message": "Missing required parameters: 224017, 54, 127522, 129293, 18293, 212038",
            "details": "ConstraintViolationException.MissingRequiredParameters",
            "path": "parameters",
            "userMessage": "Missing mandatory parameters: Manufacturer's code, Size, Color, Main Material, Sleeve, Properties."
        }
    ]
}
```
when I get offer detail again, it still was not validated

```
"validation": {
        "errors": [],
        "warnings": [],
        "validatedAt": "2023-08-28T13:22:14.263Z"
    }
```

I have also tried PATCHING external.id, delivery.handlingTime and some ather attriubes, with same exact result. So How do I trigger the offer validation properly please?

I cannot be using only https://api.allegro.pl/sale/offers/unfilled-parameters resource, as it does not cover the DRAFT offers and does not give me information on anything, but parameters.

Thanks in advance for your help

Jirka

**Comment by ZajdaBear** [comment]
Ok, as mentioned in my message, I have tried with _'external.id, delivery.handlingTime and some ather attriubes'_ with the same result. Might be easier, if you tell me what atributes to update, to trigger the validation.

Thx.

**Comment by ZajdaBear** [comment]
An updete:
I have ended the offer using PATCH https://api.allegro.pl/sale/product-offers/13591833929 
```
{
    "publication": {
        "status": "ENDED"
    }
}
```
This worked

Then I tried to activate offer again, to trigger the validation. My request looked like this
PATCH https://api.allegro.pl/sale/product-offers/13591833929 
```
    {
        "publication": {
            "status": "ACTIVE"
        }
    }
```
response was 202 Accepted

When I check the offer again, publication status is still "ENDED", which I understand, but even after this, THere are no validation errors and "validateAt" is still "2023-08-28T13:22:14.263Z" 
trace-id: b336513027683c3d

Is this a bug?

**Comment by AureliuszBrussy** [comment]
I raised this with the proper team. We are analyzing it. I will get back to you with an answer.

**Comment by MaciejFrackowiak** [comment]
Unfortunately, not yet.

**Comment by MaciejFrackowiak** [comment]
We have received confirmation that the offer price change (which you made) does not validate the offer. Also, when you finish an offer, we will not change the date in the "validatedAt" field. Delivery, taxSettings changes do not affect "validatedAt" field either.

**Comment by ZajdaBear** [comment]
Hi, thx. What does trigger the validation for my use case?

**Comment by MaciejFrackowiak** [comment]
If you change the price, delivery or taxSettings we do not change "validatedAt" field. We found that you changed the price and after that, you finished an offer. As a result, we did not change a value in the "vaildatedAt" field.

**Comment by ZajdaBear** [comment]
Let me get back to original question. How od I trigger offer validation through the API?

**Comment by MaciejFrackowiak** [comment]
Offer is validated when:
1) you activate it;
2) you change any field other than price, taxSettings or delivery.
